src/platforms/web/compiler/directives/model.js

```diff
@@ -4,6 +4,7 @@ import config from 'core/config'
 import { addHandler, addProp, getBindingAttr } from 'compiler/helpers'
 import { genComponentModel, genAssignmentCode } from 'compiler/directives/model'
 
+const ternaryTextInputRE = /^[^?]{1,}\?\s*('|"|`)(?:text|number|password|search|email|tel|url)\1\s*:\s*('|"|`)(?:text|number|password|search|email|tel|url)\2\s*$/
```

Comment:
@posva, #6344 is about compiling itself while this PR is just to remove Warning in this specific scenario.